opencl/CMakeLists.txt

```
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.14)
```

```suggestion
cmake_minimum_required(VERSION 3.11)
```
What cmake feature requires 3.14?

---

@bader `FetchContent_MakeAvailable`. Although, we can simulate it with something like
```
FetchContent_GetProperties(package)
if(NOT package_POPULATED)
  FetchContent_Populate(package)
  add_subdirectory(${package_SOURCE_DIR} ${package_BINARY_DIR})
endif()
```

or even better

```
if(${CMAKE_VERSION} VERSION_LESS 3.14)
    macro(FetchContent_MakeAvailable NAME)
        FetchContent_GetProperties(${NAME})
        if(NOT ${NAME}_POPULATED)
            FetchContent_Populate(${NAME})
            add_subdirectory(${${NAME}_SOURCE_DIR} ${${NAME}_BINARY_DIR})
        endif()
    endmacro()
endif()
```

See https://cliutils.gitlab.io/modern-cmake/chapters/projects/fetch.html

---

We use cmake_minimum_required(VERSION 3.14) for sycl/CMakeLists.txt already, so having 3.14 here should be fine imho.

---

That's right, but the community uses the lower version (3.13.4 to be exact: https://github.com/llvm/llvm-project/blob/main/llvm/CMakeLists.txt). And we uplifted it for the same reason. Anyway, I'm ok to stick to 3.14.

---

I'll mark this as resolved. Let me know if any objections.

---

3.14 is needed for FetchContent_MakeAvailable.